code/game/machinery/kitchen/smartfridge.dm

```diff
@@ -226,6 +226,40 @@
 
 	RefreshParts()
 
+/obj/machinery/smartfridge/bloodbank
+	name = "\improper Refrigerated Blood Bank"
+	desc = "A refrigerated storage unit for blood packs"
+	icon_state = "bloodbank"
+	icon_on = "bloodbank"
+
+	accepted_types = list(/obj/item/weapon/reagent_containers/blood)
+
+/obj/machinery/smartfridge/bloodbank/New()
+	. = ..()
+
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/smartfridge/bloodbank,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/matter_bin,
+		/obj/item/weapon/stock_parts/matter_bin,
+		/obj/item/weapon/stock_parts/matter_bin,
+		/obj/item/weapon/stock_parts/matter_bin,
+		/obj/item/weapon/stock_parts/scanning_module,
+		/obj/item/weapon/stock_parts/console_screen,
+		/obj/item/weapon/stock_parts/console_screen
+	)
+
+	RefreshParts()
+
+	insert_item(new /obj/item/weapon/reagent_containers/blood/APlus)
```

Comment:
This would make it spawn blood packs whenever it's constructed, so you could make blood by just constructing blood banks.

Reply:
How do I make it check if it is a roundstart smartfridge before adding the contents?

Reply:
Make a `/filled` subtype that does it, map that instead, but keep the board pointing at the empty one.

Reply:
@Exxion had a cool idea to do away with mapping subtypes and instead check to see if the ticker exists yet in New()

Why not do that?

Reply:
That's terrible because it breaks with vault loading for example.

Reply:
Can you think of an alternative we could check for?

Reply:
We could let the BYOND VM execute extra code paths by making it resolve extra type data when loading the map. It's complicated but feasible.

Reply:
That suggestion was a joke